python/delta_sharing/protocol.py

```diff
@@ -176,6 +176,24 @@ def from_json(json) -> "Protocol":
         return Protocol(min_reader_version=int(json["minReaderVersion"]))
 
 
+@dataclass(frozen=True)
+class DeltaProtocol:
```

Comment:
Could you remind me when do we need to parse the json for delta protocol/metadata? I thought all we need to do is to dump the raw json into temp files, then kernel-rust will read from those files and parse the json instead. 

I'm not familiar with kernel-rust interface though, we should define these classes only when the inner fields are needed when we use the kernel-rust API.

Reply:
Mainly for testing it helps to parse the object into JSON so that each property can be properly checked. 

I am also going to use the JSON for autoresolving the query format:
- if the response can be parsed into Protocol then we stick to original codepath
- if the response cannot be parsed into Protocol but can be parsed into DeltaProtocol we know we need to use the new codepath using kernel-rust